.changeset/giant-seas-complain.md

```diff
@@ -0,0 +1,37 @@
+---
+"@salt-ds/lab": minor
+---
+
+- Refactored `Drawer` to use floating-ui and Salt's `Scrim`.
+- Added optional `DrawerCloseButton`.
+- Implemented desktop support.
+
+```tsx
+export const DrawerTemplate = (): ReactElement => {
+  const [open, setOpen] = useState(false);
+  const id = "default-drawer";
+
+  const handleRequestOpen = () => {
+    setOpen(true);
+  };
+
+  const onOpenChange = (newOpen: boolean) => {
+    setOpen(newOpen);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Button onClick={handleRequestOpen}>Open Drawer</Button>
+      <Drawer open={open} onOpenChange={onOpenChange} id={id}>
+        <h2 id={`${id}-header`}>Title</h2>
+        <DrawerCloseButton onClick={handleClose} />
+        <p id={`${id}-content`}>Content of drawer</p>
```

Comment:
2 things:
- I'm not sure we'd actually want to label the content here. Is there a reason we are doing this?
- Why aren't we using the Salt text components

Reply:
I added both header and content to keep them in line with what we have in Overlay. Everything (id, header, content) exists only within the modal drawer, given id isn't attached to drawer's trigger element, perhaps no need for them to be set even? 

Let me update the examples to use Salt's Text